src/metrics_test.go

```diff
@@ -148,3 +158,70 @@ func Test_pathToPrefix(t *testing.T) {
 		})
 	}
 }
+
+func Test_getMetricsData(t *testing.T) {
+	tests := []struct {
```

Comment:
I prefer to have one test for each case, is more easy to read and express intention.

Reply:
At first I thought we were testing happy path more than errors handling